src/connect/planner.cpp

```diff
@@ -72,14 +73,56 @@ namespace {
         return is_on_usb && !contains_upper;
     }
 
-    bool path_exists(const char *path) {
+    bool file_exists(const char *path) {
         struct stat st = {};
         // This could give some false negatives, in practice rare (we don't have permissions, and such).
-        return stat(path, &st) == 0;
+        return stat(path, &st) == 0 and !S_ISDIR(st.st_mode);
+    }
+
+    bool dir_exists(const char *path) {
+        struct stat st = {};
+        // This could give some false negatives, in practice rare (we don't have permissions, and such).
+        return stat(path, &st) == 0 and S_ISDIR(st.st_mode);
     }
 
     template <class>
     inline constexpr bool always_false_v = false;
+
+    optional<const char *> delete_file(const char *path) {
```

Comment:
I'm not sure if `optional<char *>` isn't a bit of overcomplicated… `char *` has a `nullptr` value 😈